### PR TITLE
Add backward compatible constructor

### PR DESCRIPTION
**Pull request by ydai1124**
#1199 removed the constructor `(KafkaAvroSchemaRegistry, String)` and introduced backwards incompatible change. Restored this constructor.
@pcadabam Can you review?

### CODE REVIEW

File: gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java [hunk 1]
```diff
@@ -27,13 +27,28 @@
 public class DefaultLimiterFactory {
 
   public static final String EXTRACT_LIMIT_TYPE_KEY = "extract.limit.type";
-  public static final String EXTRACT_LIMIT_RATE_LIMIT_KEY = "extract.limit.rate.limit";
-  public static final String EXTRACT_LIMIT_RATE_LIMIT_TIMEUNIT_KEY = "extract.limit.rate.limit.timeunit";
-  public static final String EXTRACT_LIMIT_TIME_LIMIT_KEY = "extract.limit.time.limit";
-  public static final String EXTRACT_LIMIT_TIME_LIMIT_TIMEUNIT_KEY = "extract.limit.time.limit.timeunit";
+
+  public static final String EXTRACT_LIMIT_RATE_LIMIT_KEY = "extract.limit.rateLimit";
```
Comment (pcadabam) [inline gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java]: Do all these constants need to be `public`? If not make them `private` or package protected for unit tests only.

File: gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java [hunk 2]
```diff
@@ -90,4 +106,24 @@ public static Limiter newLimiter(State state) {
         throw new IllegalArgumentException("Unrecognized Limiter type: " + type.toString());
     }
   }
+
+  private static State convertDeprecatedConfigs(State state) {
```
Comment (pcadabam) [inline gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java]: Leave a comment what this does

**Comment by pcadabam** [comment]
Please add some unit tests that check initialization with both old and new config keys.

File: gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java (hunk 1 above)
Comment (ydai1124) [inline gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java]: Yes, they are used by `gobblin-runtime`.

File: gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java (hunk 2 above)
Comment (ydai1124) [inline gobblin-utility/src/main/java/gobblin/util/limiter/DefaultLimiterFactory.java]: Fixed.

**Comment by ydai1124** [comment]
@pcadabam Addressed comments.

**Comment by pcadabam** [comment]
LGTM

**Pull request merged**